src/components/materialization/source/deltaUpdates/Form.tsx

```diff
@@ -1,16 +1,32 @@
 import type { BaseFormProps } from 'src/components/shared/specPropEditor/types';
 
+import { useEffect, useRef } from 'react';
+
 import SpecPropBoolean from 'src/components/shared/specPropEditor/SpecPropBoolean';
+import { useEntityWorkflow } from 'src/context/Workflow';
 import { useSourceCaptureStore } from 'src/stores/SourceCapture/Store';
 
 export default function DeltaUpdatesForm({
     currentSetting,
     scope,
     updateDraftedSetting,
 }: BaseFormProps) {
-    const [setDeltaUpdatesHasError] = useSourceCaptureStore((state) => [
-        state.setDeltaUpdatesHasError,
-    ]);
+    const workflow = useEntityWorkflow();
+
+    const [setDeltaUpdatesHasError, setDeltaUpdates] = useSourceCaptureStore(
+        (state) => [state.setDeltaUpdatesHasError, state.setDeltaUpdates]
+    );
+
+    // If we are editing make sure we store the current value into the store "on load"
+    const defaultValue = useRef(workflow === 'materialization_edit');
+    useEffect(() => {
+        if (defaultValue.current) {
+            if (currentSetting) {
+                setDeltaUpdates(currentSetting);
+                defaultValue.current = false;
+            }
+        }
+    }, [currentSetting, setDeltaUpdates]);
```

Comment:
Make sure we load in the setting from the spec on load. If there is nothing here on load then this will fire _once_ the first time they add a setting and shouldn't cause any issues.